### ISSUE TITLE
mysqldump: Couldn't execute 'SELECT BINLOG_GTID_POS('', '0')': You are not using binary logging (1381)

### ISSUE THREAD

**Issue by cmonty14** [JGroselle/mariabackup#2]
Running mariabackup seems to work, but there's an error message in the output:
```
ct113-devwiki:/usr/local/bin# mariabackup -b bookstack
grep: bookstack: No such file or directory
2018-10-06_10:45:11 [INFO] Today backup already exist, new one is generated
2018-10-06_10:45:11 [INFO] Start backuping database bookstack
mysqldump: Couldn't execute 'SELECT BINLOG_GTID_POS('', '0')': You are not using binary logging (1381)
```

What's causing this error?

**Comment by JGroselle** [comment]
Hi, thank you for your report.
The grep error is due to the management of stderr. I will fix that.
The binlog issue is due to the option `--flush-logs`. I will add an option to set if binary logging are activated or not. By the way, you should activate them:
https://mariadb.com/kb/en/library/binary-log/

**Comment by JGroselle** [comment]
Edit: Grep is in a test `[ ... ]` so ouput is already managed.
What is your bash version?
Can you execute `[ "grep -E ".*\s.*" test" ] || echo OK`

**Comment by cmonty14** [comment]
> [ "grep -E ".*\s.*" test" ] || echo OK

```
ct113-devwiki:~# bash [ "grep -E ".*\s.*" test" ] || echo OK
/usr/bin/[: /usr/bin/[: cannot execute binary file
OK
ct113-devwiki:~# bash -v
ct113-devwiki:~# bash --version
bash --version
GNU bash, version 4.4.19(1)-release (x86_64-alpine-linux-musl)
Copyright (C) 2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

```

**Comment by cmonty14** [comment]
> Hi, thank you for your report.
> The grep error is due to the management of stderr. I will fix that.
> The binlog issue is due to the option `--flush-logs`. I will add an option to set if binary logging are activated or not. By the way, you should activate them:
> https://mariadb.com/kb/en/library/binary-log/

As of now I don't intent to replay backup logs or for replication. Therefore I didn't activate this option.
Please note that this server is running with Alpine Linux 3.8 that is a tiny linux; I want to keep things as simple as possible.

**Comment by JGroselle** [comment]
Fixed for #3 .
